### ISSUE TITLE
Feature request: Ability to choose the upper folder name

### ISSUE THREAD

**Issue by joaofl** [transmission/transmission#638]
Whenever a torrent is added, it naturally ends up in a sub-folder the user chose, but in a folder named after the torrent. 

Would be nice to have the option to rename the final destination folder before downloading the file, so renaming the folder is not required after downloading.

My reasoning for that is:

- Most of the times one download folders with names like `Movie.Name.Complete.1080p.BluRay.x264.` whereas it could simply be named `Movie Name`.
- So if you want to keep your library organized, you rename that folder, but sometimes without being sure if you have finished downloading and/or have uploaded it enough. Which results in missing files.

Said that, I believe that an option to choose the upper folder final name would be a great thing. That option would show on the same dialog that pops up when adding a torrent from a file or magnet.
In case of magnets, where the name is not yet available, it could trigger a rename action later, just before the download starts (leaving blank for default name).

I would gladly help on implementing that.

**Comment by transbot** [comment]
totally agreed with that, a feature should added a decade ago.

**Comment by ckerr** [comment]
I think I am misunderstanding this request (or maybe so much time has passed that the feature was  implemented in the interm :smile:) but this sounds like the existing feature that lets users rename a folder or filename from inside Transmission. Am I misunderstanding?

**Comment by drws** [comment]
The problem for me is that folder needs to be created and then renamed. That can already pose a file-conflict situation. Additionally, if a magnet link is downloaded, some metadata (including the name) is sometimes not known in advance. So one has to wait for the metadata to download, for the download to start and only after that the folder can be renamed. A solution would be that top-level folder name could be input in the Torrent Options ("add torrent") window in any case. It would rename a top-level torrent folder and if the magnet link only contained files, it would also put them in the named folder.

**Comment by barn4k** [comment]
Any updates on this?

**Comment by mokahless** [comment]
After #1440 was rejected, I'm glad to see this is still open but sad to see it doesn't seem to be a priority.
Came back to transmission recently and promptly left after I remembered this was one of my main frustrations. It discourages seeding once you organize new files.

So, I'd just like to help keep attention on this request and tack on a request for #1440 be included in this feature, if it is being worked on. I'd also like to clarify that #514 was closed because it was inferred it would be included in this update.

**Comment by kambala-decapitator** [comment]
this one as well as ability to not create top folder are very convenient features as some torrents use bad naming

**Comment by rapaggio** [comment]
>this one as well as ability to not create top folder are very convenient features as some torrents use bad naming

agree, that’s how it looks in qBittorrent
folder renaming + skipping(or adding) the top folder would be super handy in Transmission
really useful for me and others from what I can see

<img width="263" height="87" alt="Image" src="https://github.com/user-attachments/assets/1fbead29-7a58-4dc3-87ac-b51534776503" />
<img width="303" height="212" alt="Image" src="https://github.com/user-attachments/assets/7c260d02-8e9a-4795-a8e6-13809c4a38cc" />
